contracts/guess-the-number/src/lib.rs

```diff
@@ -30,6 +24,29 @@ impl GuessTheNumber {
     pub fn guess(env: &Env, a_number: u64) -> bool {
         a_number == env.storage().instance().get::<_, u64>(&THE_NUMBER).unwrap()
     }
+
+    /// Upgrade the contract to new wasm. Only callable by admin.
+    pub fn upgrade(env: &Env, new_wasm_hash: BytesN<32>) {
+        Self::require_admin(env);
+        env.deployer().update_current_contract_wasm(new_wasm_hash);
+    }
+
+    fn admin(env: &Env) -> Option<Address> {
+        env.storage().instance().get(ADMIN_KEY)
+    }
+
+    fn set_admin(env: &Env, admin: &Address) {
+        // Check if admin is already set
+        if env.storage().instance().has(ADMIN_KEY) {
```

Comment:
Wait, is this right? Current admin should also be able to set a new admin, right? As in the source of the admin sep

Reply:
This is keeping it simple. I chatted with Willem about it. For a real admin transfer, there should be a pending state for acceptance. This keeps it very simple for a simple contract